test/check/check.go

```diff
@@ -0,0 +1,270 @@
+}
+
+func waitForComponentRemoval(component *Component) error {
+	errs := []error{}
+	errsAppend := func(err error) {
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if component.Namespace != "" {
+		errsAppend(waitForNamespaceRemoval(component.Namespace, 4*time.Minute))
+	}
+
+	if component.ClusterRole != "" {
+		errsAppend(waitForClusterRoleRemoval(component.ClusterRole, 10*time.Second))
+	}
+
+	if component.ClusterRoleBinding != "" {
+		errsAppend(waitForClusterRoleBindingRemoval(component.ClusterRoleBinding, 10*time.Second))
+	}
+
+	if component.SecurityContextConstraints != "" {
+		errsAppend(waitForSecurityContextConstraintsRemoval(component.SecurityContextConstraints, 10*time.Second))
+	}
+
+	return errsToErr(errs)
+}
+
+func errsToErr(errs []error) error {
+	if len(errs) == 0 {
+		return nil
+	}
+	errsStrings := []string{}
+	for _, err := range errs {
+		errsStrings = append(errsStrings, err.Error())
+	}
+	return errors.New(strings.Join(errsStrings, "\n"))
+}
+
+func checkForNamespace(name string) error {
+	return framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: name}, &corev1.Namespace{})
+}
+
+func checkForClusterRole(name string) error {
+	return framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: name}, &rbacv1.ClusterRole{})
+}
+
+func checkForClusterRoleBinding(name string) error {
+	return framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: name}, &rbacv1.ClusterRoleBinding{})
+}
+
+func checkForSecurityContextConstraints(name string) error {
+	err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: name}, &securityapi.SecurityContextConstraints{})
+	if strings.Contains(err.Error(), "no kind is registered for the type") {
+		return nil
+	}
+	return err
+}
+
+func checkForDeployment(name string, namespace string) error {
+	deployment := appsv1.Deployment{}
+
+	err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, &deployment)
+	if err != nil {
+		return err
+	}
+
+	if deployment.Status.Replicas != deployment.Status.AvailableReplicas {
+		manifest, err := yaml.Marshal(deployment)
+		if err != nil {
+			panic(err)
+		}
+		return fmt.Errorf("Deployment %s/%s is not ready, current state:\n%v", namespace, name, string(manifest))
+	}
+
+	return nil
+}
+
+func checkForDaemonSet(name string, namespace string) error {
+	daemonSet := appsv1.DaemonSet{}
+
+	err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, &daemonSet)
+	if err != nil {
+		return err
+	}
+
+	if daemonSet.Status.DesiredNumberScheduled != daemonSet.Status.CurrentNumberScheduled {
+		manifest, err := yaml.Marshal(daemonSet)
+		if err != nil {
+			panic(err)
+		}
+		return fmt.Errorf("DaemonSet %s/%s is not ready, current state:\n%v", namespace, name, string(manifest))
+	}
+
+	return nil
+}
+
+func waitForNamespaceRemoval(name string, timeout time.Duration) error {
```

Comment:
We have [waitForDeletion](https://github.com/operator-framework/operator-sdk/blob/master/pkg/test/e2eutil/wait_util.go#L74) at operator-sdk testing, maybe we can use that to remove scaffolding here.